bin/MadGraph5_aMCatNLO/cards/production/13TeV/WPrime_tb/WPrime_tb_proc_card.dat

```diff
@@ -12,4 +12,4 @@ add process p p > wp+ > bs ts j, (ts > ws bs, ws > ls ls) @1
 add process p p > wp- > bs ts j, (ts > ws bs, ws > ls ls) @1
 add process p p > wp+ > bs ts j j, (ts > ws bs, ws > ls ls) @2
 add process p p > wp- > bs ts j j, (ts > ws bs, ws > ls ls) @2
-output WPrime_tb -nojpeg
+output {name} -nojpeg
```

Comment:
I'm clearly not understanding something here. Are the light quarks not given any mass in these schemes? We expect the b pdf contribution to be one of our largest cross sections. Is there any documentation that you can point me to that will help me understand these matching schemes?

Reply:
We have no diagrams without a b coupling on the production side of the W' in this model. When we built a similar model for a Z', we sought the advice of Michelangelo Mangano of how to treat the MLM matching correctly for resonances with initial state and associated b-quarks. To quote him, in two parts:
"The process you are studying is however rather peculiar, because of the initial state b's. The modeling of the initial state evolution for a b quark in the MCs is not greatly accurate, and this will be reflected in a poor matching of the samples with different multiplicity. 

For example, your 1-jet sample will likely be dominated by gb->bZ', rather than by bb->gZ'. The gb->bZ' process appears in the shower evolution of bb->Z', when the initial-state b evolves backward into a gluon that splits into b+bbar. But if the g->bb splitting is not well modeled, then you find a mismatch between the 0-jet plus shower and the 1-jet rate (you probably are aware of long-standing issues on the 4- vs 5-flavour approach to calculations involving initial-state heavy flavours, and of the systematic differences between the two approaches, unless specific scale choices are made ... what you are doing will certianly expose these issues!)"
[...]
"If you intend to combine through matching the Z', Z'+1 and Z'+2 jet samples, all processes must be included, this is not an option. Eg for Z'+1 this means that processes like
  b b-> Z' g

  b g -> Z' b

  g b -> Z' b

  should all enter...."

Similarly, we expect W' with b c~ > wp-, W'+1 with g c~ > b~ wp- and b g > c wp-, and W'+2 with g g > b~ c wp- all need to be included in madgraph to get the multiplicities correctly. Our prior experiences with not including the matching scheme are that the Z'+2 case by itself overestimates the cross section for that subprocess manifold, for example. We expect the same issue, if we try this with the W'.

Is the issue that the standard pdf used at CMS is not accounting for the b-quark mass here? If that is the case, what alternatives do we have for getting the b-pdf contribution right? We definitely cannot exclude the b from the proton.

Or should we set the b-mass to zero, in your opinion?

Reply:
I understand that the model file is inconsistent then. The theorist who built the model at our institute, years ago, did not have a specific reason for choosing either 5 flavour or 4 flavour as a standard setting, as far as I remember. I only remember him telling us that he got no definite feedback on which would be the correct choice, so he just picked one.
Given the information you're telling us, that would indicate then that this standard setting in the model would be inappropriate.

Then we need to go against the standard settings of the model, in that case, to fulfill your requirements.

Because we cannot exclude the b-pdf contributions in a model which has only cb and bt production couplings.
If we get the multiplicity wrong, because we do not correctly do the MLM matching to avoid doublecounting between different multiplicities, that will affect the outcome on any potential limit setting. If we get the relative proportions of the multiplicities correct for the price of a possibly wrong normalization, the cross section x branching ratio exclusions will still be correct, though the model interpretation in the parameter space of the couplings will be wrong.

I can only reiterate that, according to the inventor of the MLM matching scheme, we cannot generate this kind of model in a prescription not containing bottoms in the proton correctly.

Reply:
I only remember him telling us that he got no definite feedback on which would be the correct choice, so he just picked one.
> There is no correct choice between 4FS and 5FS.

Given the information you're telling us, that would indicate then that this standard setting in the model would be inappropriate.
> Standard setting in the model file is not inappropriate. Your setting is inappropriate as you are giving a syntax that shouldn't be really used with your model file.

Reply:
> the cross section x branching ratio exclusions will still be correct

Aside from everything, I strongly object to this. Your cross section cannot be correct if you are implementing incompatible bquark mass setting and proton definition. Nothing is related to model interpretations here...
There should a difference in cross sections between 4FS and 5FS when you do DY : p p > e+ e- (4FS with mb!=0 and 5FS with mb=0 : which are the correct settings). And obviously p p > e+ e- 5FS with mb!=0 is a wrong choice and will give you a wrong cross section.

Reply:
I just had some short chat with MadGraph authors.

If somebody is using 4F default UFO model and defines proton with b b~, and makes m(b) = 0, is it effectively treated as 5F scheme in MadGraph?
> no the running of alpha_s and other setup will still be setup as 4FS. even if you set all parameter the same, the code to code that evaluates the matrix-element will be different (and less efficient). We should actually forbid user to set mass to zero if they are not zero at the model level